### ISSUE TITLE
All tests fail running py.test on RaspberryPi3

### ISSUE THREAD

**Issue by E3V3A** [mitmproxy/mitmproxy#1646]
Hi,
I'm trying to make this work on a RaspberryPi3 running latest Rasbian "jessie". I've eventually managed to install everything properly (without failures) but in the final step of running the `py.test` I get all errors. 

```bash
(venv3) pi@raspberrypi:~/Tools/mitmproxy $ py.test
=========================================================== test session starts ============================================================
platform linux -- Python 3.4.2, pytest-3.0.3, py-1.4.31, pluggy-0.4.0
rootdir: /home/pi/Tools/mitmproxy, inifile: setup.cfg
plugins: xdist-1.15.0, timeout-1.0.0, faulthandler-1.3.0, cov-2.4.0
collected 10 items / 85 errors

================================================================== ERRORS ==================================================================
___________________________________________ ERROR collecting test/mitmproxy/test_addonmanager.py ___________________________________________
ImportError while importing test module '/home/pi/Tools/mitmproxy/test/mitmproxy/test_addonmanager.py'.
Original error message:
'No module named 'typing''
Make sure your test modules/packages have valid Python names.
______________________________________________ ERROR collecting test/mitmproxy/test_certs.py _______________________________________________
ImportError while importing test module '/home/pi/Tools/mitmproxy/test/mitmproxy/test_certs.py'.
Original error message:
'No module named 'typing''
Make sure your test modules/packages have valid Python names.
...
```

I have no idea what is the problem and I also have no idea what are the correct dependencies and requirements. They are very vague at best. The project would certainly benefit from better installation instructions and clearly state the various dependencies. For example , there is no place mentioned that `Pillow` fails because, `wheel` fails, because `libjpeg-dev` is not mentioned as a dependency, nor automatically installed. There are several of these similar issues. 

So what am I missing to get these tests working?

**Comment by cortesi** [comment]
Hi there. It looks like the issue is that you're using Python3.4 - we support 3.5 only for now. 

Platform install instructions are a constant bugbear for us. They differ not only between distributions and operating systems, but also between different versions of these. Maintaining an exhaustive set of steps to go from zero to complete on every platform would be a nightmare. Our strategy instead is to have really good binary distributions and to co-operate with platform packagers where we can (i.e. the Debian & Ubuntu packages).

**Comment by mhils** [comment]
@cortesi: Shall we add a check in main.py for <3.5?

**Comment by cortesi** [comment]
@mhils Yes, I think that makes sense. Probably in version_check.py, with the ``check_pyopenssl_version`` function generalised to just ``check_versions``.

**Comment by E3V3A** [comment]
@cortesi @mhils  Hi! Thanks for looking into this. What exactly are the parts requiring **3.5**? It seem that perhaps it's not very much, since I actually got it to compile at 3.4, but then the `py.test` would fail. Perhaps I was just missing some dependencies for those tests? Then in the end I actually installed 3.5 and of course it compiled fine. The problem is having to compile 3.5 from a vanilla RPi Rasbian, which was far from trivial without online help. I also did mess up pip a few times, requiring re-installation etc.

**Comment by E3V3A** [comment]
Also the `openssl` dependency seem too recent...
```
test/mitmproxy/test_dump.py ..ALPN support missing (OpenSSL 1.0.2+ required)!
HTTP/2 is disabled. Use --no-http2 to silence this warning.
ALPN support missing (OpenSSL 1.0.2+ required)!
HTTP/2 is disabled. Use --no-http2 to silence this warning.
ALPN support missing (OpenSSL 1.0.2+ required)!
HTTP/2 is disabled. Use --no-http2 to silence this warning.
............
```
The currently stable apt version for `openssl` is:
```
openssl/stable,now 1.0.1t-1+deb8u5 armhf [installed,automatic]
  Secure Sockets Layer toolkit - cryptographic utility
```

The `test.py` takes ~8 minutes. It might be informative to the users to know about the approximate run times when running on RPi's.

**Comment by Kriechi** [comment]
@E3V3A the tests should pass with old & newer versions of openssl.
I think you can ignore these warnings, as long as the test does NOT fail.
